src/plugins/intel_cpu/src/nodes/eltwise.cpp

```diff
@@ -294,7 +295,7 @@ struct jit_uni_eltwise_generic : public jit_uni_eltwise_kernel, public jit_gener
                     this, p->entry_[i], vmm_d_weights, vmm_d_bias, reg_d_weights, reg_d_bias));
         }
 
-        if (mayiuse(avx512_core))
+        if (mayiuse(avx512_core) || mayiuse(avx2_vnni_2))
```

Comment:
if this condition fails, `uni_vcvtneps2bf16` will be empty, but `store_vector()` methods calls it unconditionally which will segment-fault. The reason why this didn't happen in practice is that the `inference_precision` only allows to be bf16 on avx512_cores, in this PR you extended bf16 inference_precision to `avx2_vnni_2` so the change is OK, but my point is, `jit_uni_vcvtneps2bf16` can actually support any existing platform (from sse41 to avx512), and the condition in line 298 should be totally removed, and uni_vcvtneps2bf16 should be always initialized. (it doesn't need to relying on other logics to prevent uninitialized `uni_vcvtneps2bf16` issue).

I saw similar problems in normalize.cpp/reduce.cpp/region_yolo.cpp/softmax.cpp, maybe we should change them all (just search jit_uni_vcvtneps2bf16 under cpu plugin src).

Reply:
Agree with you that functionally remove such kind of condition check(uni_vcvtneps2bf16 always initialized) seems better. But this will change many platforms' previous behaviour(add extra uni_vcvtneps2bf16  initialization) which may bring in performance regression issue(although this initialization time cost seems quite little). For empty instance segment-fault concern, I think we can prevent this by always keep uni_vcvtneps2bf16 usage align with inference_precision support list

Reply:
maybe you are right